internal/services/logic/logic_app_action_http_resource.go

```diff
@@ -303,6 +303,12 @@ func expandLogicAppActionHttpHeaders(headersRaw map[string]interface{}) (*map[st
 }
 
 func expandLogicAppActionHttpQueries(queriesRaw map[string]interface{}) (*map[string]string, error) {
+	// per the issue https://github.com/hashicorp/terraform-provider-azurerm/issues/28429
+	// empty map and nil are different on the service side.
+	if len(queriesRaw) == 0 {
+		return nil, nil
+	}
+
```

Comment:
@ziyeqf could you extend the following test case to add a second step where `queries` is removed and validate that this works for that case?
https://github.com/hashicorp/terraform-provider-azurerm/blob/f6886d7b22e8f805e561b6fc36214432194719a8/internal/services/logic/logic_app_action_http_resource_test.go#L82-L95